### PR TITLE
Allow test cases to support commas. #689

### PR DESCRIPTION
**Pull request by TimothyJCowen**
Will help make an upcoming issue more clear.

Closes https://github.com/Brightspace/D2L.CodeStyle/issues/689

### CODE REVIEW

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 1]
```diff
@@ -338,7 +338,10 @@ ImmutableArray<string> arguments
 
 					yield return new DiagnosticExpectation(
 						name: name,
-						arguments: arguments.Split( ',' ).Select( RemoveSingleLeadingSpace ).ToImmutableArray()
+						arguments: Regex.Split( arguments, @"(?<!\\\\)," )
+						.Select( arg => arg.Replace( "\\\\", string.Empty ) )
+						.Select( RemoveSingleLeadingSpace )
+						.ToImmutableArray()
```
Comment (omsmith) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:344]: nit

```suggestion
							.Select( arg => arg.Replace( "\\\\", string.Empty ) )
							.Select( RemoveSingleLeadingSpace )
							.ToImmutableArray()
```

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 2]
```diff
@@ -338,7 +338,10 @@ ImmutableArray<string> arguments
 
 					yield return new DiagnosticExpectation(
 						name: name,
-						arguments: arguments.Split( ',' ).Select( RemoveSingleLeadingSpace ).ToImmutableArray()
+						arguments: Regex.Split( arguments, @"(?<!\\\\)," )
+							.Select( arg => arg.Replace( "\\\\", string.Empty ) )
```
Comment (omsmith) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]: I'm struggling to wrap my head around whether this should actually be `\\\\` or `\\`. Can you share an example of an expectation you've got working with this?

Comment (omsmith) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]:
```suggestion
						arguments: Regex.Split( arguments, @"(?<!\\\\)," )
							.Select( arg => arg.Replace( "\\\\,", "," ) )
```

Comment (TimothyJCowen) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]: I've tested with this and it works perfectly:
```cs
/* NonImmutableTypeHeldByImmutable(interface, SpecTests.Types.SomeGenericInterface<int\\, int>, ) */
```

Comment (TimothyJCowen) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]: Is the reason for this just to make it only catch escaped commas?

Comment (omsmith) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]: Right

Comment (omsmith) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:342]: So given it's a comment, you don't actually need to be escaping the `\`. The following would be fine, and the regexes switched to `\\`. Sound reasonable?

`/* NonImmutableTypeHeldByImmutable(interface, SpecTests.Types.SomeGenericInterface<int\, int>, ) */`

**Review by omsmith — APPROVED** [review]

**Pull request merged**